instaloader/structures.py

```diff
@@ -553,10 +553,14 @@ def location(self) -> Optional[PostLocation]:
             return None
         location_id = int(loc['id'])
         if any(k not in loc for k in ('name', 'slug', 'has_public_page', 'lat', 'lng')):
-            loc = self._context.get_json("explore/locations/{0}/".format(location_id),
-                                         params={'__a': 1})['graphql']['location']
-        self._location = PostLocation(location_id, loc['name'], loc['slug'], loc['has_public_page'],
-                                      loc['lat'], loc['lng'])
+            loc_post_requested = self._context.get_json("explore/locations/{0}/".format(location_id),
+                                         params={'__a': 1})['native_location_data']['location_info']
+        loc_name = loc.get("name") or loc_post_requested.get("name")
+        loc_slug = loc.get("slug") or loc_post_requested.get("slug")
+        loc_has_public_page = loc.get("has_public_page") or loc_post_requested.get("has_public_page")
+        loc_lat = loc.get("lat") or loc_post_requested.get("lat")
+        loc_lng = loc.get("lng") or loc_post_requested.get("lng")
+        self._location = PostLocation(location_id, loc_name, loc_slug, loc_has_public_page, loc_lat, loc_lng)
```

Comment:
Seems to be equivalent to (less code):

```suggestion
            loc.update(self._context.get_json("explore/locations/{0}/".format(location_id),
                                              params={'__a': 1})['native_location_data']['location_info'])
        self._location = PostLocation(location_id, loc['name'], loc['slug'], loc['has_public_page'],
                                      loc['lat'], loc['lng'])
```